src/app/shared/services/query.service.ts

```diff
@@ -0,0 +1,82 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { Headers, Http, RequestOptions } from '@angular/http';
+
+@Injectable()
+export class QueryService {
+  options: any
+  constructor(private http: Http) {
+    document.cookie = "_pk_id.11.1fff=99911b90596f9e8d.1504879999.2.1505216503.1505216503.; _pk_ses.11.1fff=*"
+    this.options = this.createAuthorizationHeader()
+   }
+  
+   createAuthorizationHeader() {
+    let headers = new Headers()
+    headers.append('Access-Control-Allow-Origin', 'localhost:4200');
+    headers.append('Access-Control-Allow-Credentials', 'true');
+    let options = new RequestOptions({ headers: headers, withCredentials: true });
+    return options
+  }
+
+  submitActions(userActions:object[], url:string, checkedRecords:number[], allSelected:boolean): Promise<Object>{
+    return this.http
+    .post(`${url}/update`, {
```

Comment:
for object literal do it,
`{ foo: bar, dude, prop: aVariable}` instead of `{ 'foo': bar, 'dude': dude, 'prop': aVariable}`